### DISCUSSION TITLE
Build issues in libmesh: Undefined reference to HDF components

### DISCUSSION THREAD

**Discussion by abarun22** [idaholab/moose#21336] [Q&A General] [answered]
Hello all,
Strangely the build of libmesh throws the following undefined references even after the inclusion of HDF modules in my environment.
```
/home/abalasub/Working/Projects/moose/libmesh/build/contrib/netcdf/v4/liblib/.libs/libnetcdf.so: undefined reference to `H5Pset_all_coll_metadata_ops'
/home/abalasub/Working/Projects/moose/libmesh/build/contrib/netcdf/v4/liblib/.libs/libnetcdf.so: undefined reference to `H5Pset_dxpl_mpio'
/home/abalasub/Working/Projects/moose/libmesh/build/contrib/netcdf/v4/liblib/.libs/libnetcdf.so: undefined reference to `H5Pset_coll_metadata_write'
/home/abalasub/Working/Projects/moose/libmesh/build/contrib/netcdf/v4/liblib/.libs/libnetcdf.so: undefined reference to `H5Pset_fapl_mpio'
/home/abalasub/Working/Projects/moose/petsc/arch-moose/lib/libpetsc.so: undefined reference to `H5Pget_dxpl_mpio'
```
We have hdf5 versions 1.8.13 and 1.12 - inclusion of both of them did not help solve this problem. This is noticed only in the recent sources of MOOSE and not in the older versions. Any references here will be much appreciated.
Kind regards,
Arun

**Comment by GiudGiud** [comment, marked as answer]
Hello

How are you including these HDF5 libraries? Are you setting an environment variable?
Is there ANY other HDF5 libraries that are loaded at the same time?

I'd wipe the libmesh directory once to make sure there is not some linking to an old HDF5 library around

Guillaume

**Comment by abarun22** [comment]
Hi Guillaume,
I was including the HDF5 libraries in my environment through the modules. I
had to initially remove the inclusion of HDF5 during petsc build due to the
fact that it was not passing the build checks. When I tried to build
libmesh further it throwed undefined references to HDF5 components. I
thought I could reinstate the HDF5 libraries this time and see if I can
build both petsc and libmesh. It eventually builds OK and also passes the
tests. I've no idea why it did not pass petsc build initially.
In any case it works fine.
Kind regards,
Arun

**Reply by GiudGiud** [reply]
Ah that's great news. HDF5 can be a little finicky, one definitely has to be extremely consistent throughout the build, same as for compilers.
